src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int COUNT_OF_FIGURE = 5;
+    private static final String DEFAULT_COLOR = "white";
+    private static final double DEFAULT_RADIUS = 10;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        switch (random.nextInt(COUNT_OF_FIGURE)) {
+            case 0: return createCircle();
+            case 1: return createRightTriangle();
+            case 2: return createSquare();
+            case 3: return createRectangle();
+            case 4: return createIsoscelesTrapezoid();
+            default: return createSquare();
+        }
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(DEFAULT_COLOR, DEFAULT_RADIUS);
+    }
+
+    private Figure createCircle() {
+        return new Circle(colorSupplier.getRandomColor(), random.nextDouble());
```

Comment:
passing expression as constructor arguments decreases the overall readability

 I would suggest declaring all the necessary variables and then pass them into constructor

see [here](https://mate-academy.github.io/jv-program-common-mistakes/java-core/abstract-class-interface/oop-advanced#%D1%81reating-a-figure-dont-pass-expressions-in-constructor)